app/models/series.rb

```diff
@@ -950,27 +950,25 @@ def Series.assign_dependency_depth
     previous_depth_count = Series.where(universe: 'UHERO', dependency_depth: 0).count
 
     # first level of dependencies
-    first_level_sql = <<~SQL
+    ActiveRecord::Base.connection.execute(<<~SQL)
       UPDATE series s SET dependency_depth = 1
-      WHERE EXISTS (SELECT 1 FROM data_sources WHERE `dependencies` LIKE CONCAT('% ', s.`name`, '%'));
+      WHERE EXISTS (SELECT 1 FROM data_sources WHERE dependencies LIKE CONCAT('% ', s.name, '%'));
     SQL
-    ActiveRecord::Base.connection.execute(first_level_sql)
     current_depth_count = Series.where(universe: 'UHERO', dependency_depth: 1).count
 
     previous_depth = 1
-    until current_depth_count == previous_depth_count
```

Comment:
It's possible for us to have a circular dependency (at least they have existed in the past). In those cases, `previous_depth_count` equals `current_depth_count` once we have eliminated the non-circular dependencies. We could add ` || current_depth_count == 0` to the original `until`.

Reply:
If you can guarantee no circular dependencies (i.e., that we will in fact converge to `current_depth_count == 0`, we can go with your solution.